test/pylib/s3_proxy.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+# S3 proxy server to inject retryable errors for fuzzy testing.
+
+import os
+import random
+import sys
+import asyncio
+import requests
+import threading
+from http.server import BaseHTTPRequestHandler, ThreadingHTTPServer
+from urllib.parse import urlparse, parse_qs
+import uuid
+from functools import partial
+from collections import OrderedDict
+from requests import Response
+from typing_extensions import Optional
+
+sys.path.insert(0, os.path.dirname(__file__))
+
+
+class Policy:
+    def __init__(self, max_retries: int):
+        self.should_forward: bool = True
+        self.should_fail: bool = False
+        self.error_count: int = 0
+        self.max_errors: int = random.choice(list(range(1, max_retries)))
```

Comment:
not sure if it's worthy of adding a command line option of a setting for setting the seed for this random test. i think it might be helpful to reproduce a bug identified by this test.

Reply:
you want to set some max number to keep the client succeeding, once you change the max retries in the client the proxy may fail on exhausting all retries, then you can just change the command line to keep it working as expected

Reply:
@kreuzerkrieg i don't think your reply is related to my suggestion. probably i failed to explain my suggestion in an understandable way: this is a random test, to make it repeatable, we can add a command line argument to add the *seed*.

Reply:
right, completle misunderstood your comment, lemme check how do I make it re-playable

Reply:
Done